### PR TITLE
Added possibility to use TLS with systemd socket activation

### PR DESCRIPTION
**Pull request by kayrus**
Before docker daemon used insecure connection with systemd socket activation even if you specify `--tlsverify` flag. Now it is fixed.

### CODE REVIEW

File: vendor/src/github.com/coreos/go-systemd/activation/listeners.go [hunk 1]
```diff
@@ -24,12 +25,16 @@ import (
 // The order of the file descriptors is preserved in the returned slice.
 // Nil values are used to fill any gaps. For example if systemd were to return file descriptors
 // corresponding with "udp, tcp, tcp", then the slice would contain {nil, net.Listener, net.Listener}
-func Listeners(unsetEnv bool) ([]net.Listener, error) {
+func Listeners(unsetEnv bool, tlsConfig *tls.Config) ([]net.Listener, error) {
```
Comment (jfrazelle) [inline vendor/src/github.com/coreos/go-systemd/activation/listeners.go]: this is changing the vendored files this is not how we update vendors this should be pushed upstream

Comment (kayrus) [inline vendor/src/github.com/coreos/go-systemd/activation/listeners.go]: Ok, will do,

**Comment by kayrus** [comment]
Upstream listeners.go [here](https://github.com/coreos/go-systemd/pull/121)

**Comment by kayrus** [comment]
@jfrazelle updated commit

**Comment by runcom** [comment]
could you please split the vendor commit from the fix commit?

ping @LK4D4

**Comment by kayrus** [comment]
@runcom done

**Comment by kayrus** [comment]
@thaJeztah fixed

**Comment by thaJeztah** [comment]
thanks!

ping @runcom @jfrazelle PTAL

**Comment by LK4D4** [comment]
LGTM

**Comment by runcom** [comment]
LGTM

**Pull request merged**